### PR TITLE
Better protocol upgrade api + blocking dialog

### PR DESCRIPTION
**Pull request by baptiste-grob**
- Makes SNAlertService a type which clients have to pass an implementation of in the constructor
- Adds `blockingDialog` method to SNAlertService
- Lets clients clearly know if the protocol upgrade succeeded, failed, or canceled.
- Shows a blocking dialog during the encryption upgrade process (likely needs a better message)

### CODE REVIEW

File: lib/application.ts [hunk 1]
```diff
@@ -683,10 +689,11 @@ export class SNApplication {
     return !isNullOrUndefined(this.getUser()) || this.hasPasscode();
   }
 
-  /**
-   * @returns An array of errors, if any.
-   */
-  public async upgradeProtocolVersion() {
+  public async upgradeProtocolVersion(): Promise<{
+    success?: true,
```
Comment (mobitar) [inline lib/application.ts:693]: Wait why does this default to true?

File: lib/services/api/messages.ts [hunk 2]
```diff
@@ -45,6 +45,7 @@ export const INVALID_PASSWORD                      = `Invalid password.`;
 
 export const OUTDATED_PROTOCOL_ALERT_TITLE         = 'Update Recommended';
 export const OUTDATED_PROTOCOL_ALERT_IGNORE        = 'Sign In';
+export const UPGRADING_ENCRYPTION                  = 'Upgrading the account encryption. Do not close the application.';
```
Comment (mobitar) [inline lib/services/api/messages.ts:48]: "Your account's encryption version is being upgraded. Do not close the application until this process completes."

File: lib/application.ts (hunk 1 above)
Comment (baptiste-grob) [reply lib/application.ts:693]: this isn't a default, it's the `true` type. So this says that if `success` is present, it's going to be `true`. an alternative would have been to have the success property be mandatory, and a boolean. But it just resulted in noise wherever we would `return { ... }`.

**Comment by baptiste-grob** [comment]
Heads up @radko93, this is a breaking change, you'll have to update your SNApplication constructor call and add/change a couple of things in your AlertService. Feel free to reach out if anything is left unclear after looking at the new types :)

File: lib/application.ts (hunk 1 above)
Comment (mobitar) [reply lib/application.ts:693]: Ah interesting. So TS would complain if you tried to return `{success: false}`?

Comment (baptiste-grob) [reply lib/application.ts:693]: Yes, exactly!

**Pull request merged**